### PR TITLE
[PAY-405] Fixed Claim Rewards Pill To Respect Optimizely Config

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Fixes a bug where the "Claim Rewards" Pill in the left navbar on web client would turn green whenever the user completed a challenge - even if the challenge wasn't enabled in Optimizely.

The useRewardIds hook is ripe for refactoring but I can open a separate ticket for that.

### How Has This Been Tested?
* Uploaded 3 tracks to complete the upload challenge
* De-whitelisted the upload challenge in Optimizely
* Confirmed that the pill does not turn green.
* Confirmed that it does when the upload challenge is re-enabled.
Screen Shot 2022-07-07 at 7 46 14 PM
Screen Shot 2022-07-07 at 7 48 13 PM

### CODE REVIEW

File: packages/web/src/components/nav/desktop/NavAudio.tsx [hunk 1]
```diff
@@ -27,6 +30,31 @@ const messages = {
...
+])
+
+/** Pulls rewards from remoteconfig */
+const useRewardIds = (
+  hideConfig: Partial<Record<ChallengeRewardID, boolean>>
+) => {
+  const rewardsString = useRemoteVar(StringKeys.CHALLENGE_REWARD_IDS)
+  if (rewardsString === null) return []
+  const rewards = rewardsString.split(',') as ChallengeRewardID[]
+  const filteredRewards: ChallengeRewardID[] = rewards.filter(
```
Comment (dylanjeffers) [inline packages/web/src/components/nav/desktop/NavAudio.tsx:52]: maybe `activeRewards`? and the hook could be `useActiveRewardIds`?

File: packages/web/src/components/nav/desktop/NavAudio.tsx [hunk 2]
```diff
@@ -43,8 +71,12 @@ const NavAudio = () => {
...
 
-  const userChallenges = useSelector(getOptimisticUserChallenges)
-  const hasClaimableTokens = Object.values(userChallenges).some(
+  const optimisticUserChallenges = useSelector(getOptimisticUserChallenges)
+  const rewardIds = useRewardIds({})
+  const filteredUserChallenges = Object.values(
+    optimisticUserChallenges
+  ).filter(challenge => rewardIds.includes(challenge.challenge_id))
+  const hasClaimableTokens = filteredUserChallenges.some(
     challenge => challenge && challenge.claimableAmount > 0
```
Comment (dylanjeffers) [inline packages/web/src/components/nav/desktop/NavAudio.tsx:80]: is challenge nullable here? guessing it shouldn't be after the filter?

File: packages/web/src/components/nav/desktop/NavAudio.tsx [hunk 3]
```diff
@@ -27,6 +30,31 @@ const messages = {
...
+  'connect-verified',
+  'listen-streak',
+  'profile-completion',
+  'referred',
+  'send-first-tip'
+])
+
+/** Pulls rewards from remoteconfig */
+const useRewardIds = (
+  hideConfig: Partial<Record<ChallengeRewardID, boolean>>
```
[the remark this answers is not in the document]
Comment (dharit-tan) [reply packages/web/src/components/nav/desktop/NavAudio.tsx:47]: Actually in this case we don't even need it.

File: packages/web/src/components/nav/desktop/NavAudio.tsx (hunk 1 above)
Comment (dharit-tan) [reply packages/web/src/components/nav/desktop/NavAudio.tsx:52]: Good points - I copied this over from pages/audio-rewards-page/ChallengeRewardsTile.tsx so didn't look too closely at it.

File: packages/web/src/components/nav/desktop/NavAudio.tsx (hunk 2 above)
Comment (dharit-tan) [reply packages/web/src/components/nav/desktop/NavAudio.tsx:80]: That seems right. And just checked getOptimisticUserChallenges, seems like it filters out nullable values before returning.

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-PAY-405

File: packages/web/src/components/nav/desktop/NavAudio.tsx [hunk 4]
```diff
@@ -27,6 +30,29 @@ const messages = {
...
+  'profile-completion',
+  'referred',
+  'send-first-tip'
+])
+
+/** Pulls rewards from remoteconfig */
+const useActiveRewardIds = () => {
+  const rewardsString = useRemoteVar(StringKeys.CHALLENGE_REWARD_IDS)
+  if (rewardsString === null) return []
+  const activeRewards = rewardsString.split(',') as ChallengeRewardID[]
```
Comment (dylanjeffers) [inline packages/web/src/components/nav/desktop/NavAudio.tsx:49]: Total nit, but was meaning this is rewards, and then filteredRewards is activeRewards!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-PAY-405

**Pull request merged**